Sources/Provider/Provider.swift

```diff
@@ -76,4 +78,22 @@ public protocol Provider: Sendable {
     ///   - requestBehaviors: Actions to perform before the network request is performed and / or after the network request is completed. Only called if the items weren’t successfully retrieved from persistence.
     /// - Returns: The items or error which occurred.
     func asyncProvideItems<Item: Providable>(request: any ProviderRequest, decoder: ItemDecoder, providerBehaviors: [ProviderBehavior], requestBehaviors: [RequestBehavior]) async -> Result<[Item], ProviderError>
+    
+    /// Returns a stream of a single item.
```

Comment:
Let's go ahead and remove the old async version that can crash in favor of these new ones.